src/hotspot/share/opto/superword.cpp

```diff
@@ -662,7 +662,7 @@ void SuperWord::find_adjacent_refs() {
         }
       }
     } else {
-      if (same_velt_type(mem_ref, best_align_to_mem_ref)) {
+      if (_phase->C->get_alias_index(mem_ref->adr_type()) == _phase->C->get_alias_index(best_align_to_mem_ref->adr_type())) {
```

Comment:
Would it not be nicer to wrap this in a separate function?
`same_memory_slice(mem_ref, best_align_to_mem_ref)`
You can then reuse it everywhere, makes it more readable.

You should also change the comments below, and elsewhere, as it is now not about "same type", but "same memory slice".